src/globals.c

```diff
@@ -318,25 +310,48 @@ bool init_globals(void)
 		return false;
 	}
 
-/* Set configuration defaults based on options provided
-	at configure time.  */
+/* Set defaults with following order of preference:
+	1. Runtime flags
+	2. State dir configuration files
+	3. Configure time settings
+*/
+	if (!set_format_string(NULL)) {
```

Comment:
Without looking through the code, it's hard to tell why we're doing a set_format_string(NULL) and then dropping down into the baked in configure time setting. Maybe a comment stating that the NULL param call is actually saying "First try to apply the default config file path and do not pass any values in." Otherwise the code looks ok with the new priority changes.